src/components/Leaflet/LeafletSearchInput/LeafletSearchInput.tsx

```diff
@@ -1,182 +1,236 @@
-                    ref={suggestList}>
-                    {queryData.map((item: any, index) => (
-                        <li
-                            tabIndex={0}
-                            className={`px-5 hover:underline hover:bg-gray-100 focus:underline focus:bg-gray-100 focus:ring`}
-                            key={index}
-                            data-index={index + 1}
-                            onClick={() =>
-                                locatieServerLookupQuery(
-                                    item.id,
-                                    item.weergavenaam
-                                )
+                    newIndex = parseInt(currentIndex) + 1
+                    break
+                case 'previous':
+                    if (parseInt(currentIndex) === 1) {
+                        typeof ref !== 'function' && ref?.current?.select()
+                        return
+                    }
+                    newIndex = parseInt(currentIndex) - 1
+                    break
+                default:
+                    newIndex = 1
+            }
+
+            const result = document.querySelectorAll(
+                `[data-index='${newIndex}']`
+            )[0] as HTMLLIElement
+            result.focus()
+        }
+
+        return (
+            <div ref={searchContainer}>
+                <div className="relative">
+                    <input
+                        className={
+                            classes ||
+                            'w-64 h-8 px-5 py-3 text-sm leading-tight text-gray-700 border-none rounded-sm appearance-none focus:outline-none'
+                        }
+                        type="text"
+                        ref={ref}
+                        placeholder={placeholder || 'Zoeken op de kaart'}
+                        onChange={handleChange}
+                        onFocus={() => setShowSuggestions(true)}
+                        value={searchQuery}
+                        onKeyDown={e => {
+                            if (e.key === 'ArrowDown' && queryData.length > 0) {
+                                const result = document.querySelectorAll(
+                                    `[data-index='1']`
+                                )[0] as HTMLLIElement
+                                result.focus()
                             }
-                            onKeyDown={e => {
-                                if (e.key === 'Enter') {
+                        }}
+                    />
+                    {withSearchIcon && (
+                        <FontAwesomeIcon
+                            className="inline-block text-sm absolute top-3 right-3"
+                            icon={faSearch}
+                        />
+                    )}
+                </div>
+                {showSuggestions && queryData.length > 0 ? (
+                    <ul
+                        id="searchQueryResults"
```

Comment:
Right now this has overflow: scroll, but I think it would be cleaner to set this to auto. To hide the scrollbar if there is none necessary (especially for the horizontal bar)